pkg/tests/tasks/traffic/ingress/gatewayapi_test.go

```diff
@@ -0,0 +1,164 @@
+        spec:
+          runtime:
+            components:
+              pilot:
+                container:
+                  env:
+                    PILOT_ENABLE_GATEWAY_API: “true”
+                    PILOT_ENABLE_GATEWAY_API_STATUS: “true”
+                    PILOT_ENABLE_GATEWAY_API_DEPLOYMENT_CONTROLLER: “true”`)
+			t.Cleanup(func() {
+				oc.Patch(t, meshNamespace, "smcp", smcpName, "json",
+					`[{"op": "remove", "path": "/spec/runtime"}]`)
+			})
+
+		} else {
+			oc.Patch(t, meshNamespace, "smcp", smcpName, "merge", `
+        spec:
+          techPreview:
+            gatewayAPI:
+              enabled: true`)
+
+			t.Cleanup(func() {
+				oc.Patch(t, meshNamespace, "smcp", smcpName, "json",
+					`[{"op": "remove", "path": "/spec/techPreview"}]`)
+			})
+		}
+
+		t.LogStep("Deploy the Gateway API configuration including a single exposed route (i.e., /get)")
+		oc.ApplyString(t, ns, GetGatewayAPI)
+		t.Cleanup(func() {
+			oc.DeleteFromString(t, ns, GetGatewayAPI)
+		})
+
+		t.LogStep("Verify the deployed gatewayApi")
+		shell.Execute(t,
+			fmt.Sprintf("oc wait --for=condition=Ready -n %s gateways.gateway.networking.k8s.io gateway", ns),
+			assert.OutputContains("condition met",
+				"Gateway is running",
+				"Gateway is not running"))
+
+		t.LogStep("Verfiy the GatewayApi access the httpbin service using curl")
+		retry.UntilSuccess(t, func(t TestHelper) {
+			oc.Exec(t,
+				pod.MatchingSelector("app=istio-ingressgateway", meshNamespace),
+				"istio-proxy",
+				"curl http://gateway.foo.svc.cluster.local:8080/get -H Host:httpbin.example.com -s -o /dev/null -w %{http_code}",
+				assert.OutputContains("200",
+					"Access the httpbin service with GatewayApi",
+					"Unable to access the httpbin service with GatewayApi"))
+		})
+	})
+}
+
+const GetGatewayAPI = `
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: gateway
+  namespace: foo
+spec:
+  gatewayClassName: istio
+  listeners:
+  - name: default
+    hostname: "*.example.com"
+    port: 8080
+    protocol: HTTP
+    allowedRoutes:
+      namespaces:
+        from: All
+---
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: HTTPRoute
+metadata:
+  name: http
+  namespace: foo
+spec:
+  parentRefs:
+  - name: gateway
+    namespace: foo
+  hostnames: ["httpbin.example.com"]
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /get
+    backendRefs:
+    - name: httpbin
+      port: 8000`
+
+const gatewayDeployment = `
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: gateway
+  namespace: foo
+spec:
+  selector:
+    matchLabels:
+      istio: ingressgateway
+  template:
+    metadata:
+      annotations:
+        inject.istio.io/templates: gateway
+      labels:
+        istio: ingressgateway
+        sidecar.istio.io/inject: "true"
+        istio.io/gateway-name: gateway
+    spec:
+      containers:
+        - name: istio-proxy
+          image: auto
+`
```

Comment:
In both tests, the deployment controller is enabled, so you don't have to deploy istio-proxy. Btw. attaching gateways to custom deployment is not supported by the gateway injection template, so you don't need it anyway.
```suggestion
```

Reply:
@jewertow: without deploying the gateway, it does not create the gateway API pod.  

When we deploy this gateway, the connection will fail to connect
```
apiVersion: gateway.networking.k8s.io/v1beta1
kind: Gateway
metadata:
  name: gateway
  namespace: istio-ingress
spec:
  gatewayClassName: istio
  listeners:
  - name: default
    hostname: "*.example.com"
    port: 80
    protocol: HTTP
    allowedRoutes:
      namespaces:
        from: All

```

I will deploy it on the cluster, will show you.

Reply:
@jewertow: It got successfully created the gateway deployment pod with the gateway_api. 

I will remove this yaml, thanks for checking.